service/storage_proxy.cc

```diff
@@ -990,6 +1003,14 @@ class storage_proxy::remote {
 
         auto schema = co_await get_schema_for_read(cmd.schema_version, src_addr, src_shard, *timeout);
         dht::token token = dht::get_token(*schema, key);
+
+        // This guard ensures that tablet migrations wait for replica requests,
+        // even if the LWT coordinator is no longer available.
+        // This can happen if the LWT coordinator was fenced out, a request timeout was reached on it
+        // or the coordinator node crashed and was added to the ignored_nodes list.
+        locator::token_metadata_guard guard(schema->table(), token);
+        co_await _sp.apply_fence(fence_opt, src_addr);
+
```

Comment:
We could avoid double call to `_sp.apply_fence` if we took the guard before the previous `_sp.apply_fence`.

There's a get_schema_for_read() call, but it's with a timeout, so shouldn't extend the wait beyond that.

Reply:
Well, it's on the hot path, and those cheap things add up.

> We don't want to call get_schema_for_read for stale requests.

Why not?